packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/IntentSender.java

```diff
@@ -158,7 +185,46 @@ Intent buildIntent(
         intent.setPackage(null);
       }
     }
+    Boolean isPackageFilterRequired = ignoredPackages != null && ignoredPackages.size() > 0;
```

Comment:
nit: the primitive `boolean` would be preferable here, confusingly `Boolean` is an object and could theoretically be null.